From 26928128cb33ab933d3f40ab873757322bc24ad4 Mon Sep 17 00:00:00 2001
From: Leonid Komarianskyi <leonid_komarianskyi@epam.com>
Date: Mon, 27 May 2024 10:13:41 +0300
Subject: [PATCH 2/4] gen5: vdk: switch to PL011

Currently HSCIF works slow on VDK 3.1.0 and leads to boot speed
issues. Switching to PL011 increases boot speed in approximately
up to 50 times compared with HSCIF.

Signed-off-by: Leonid Komarianskyi <leonid_komarianskyi@epam.com>
---
 arch/arm/dts/r8a78000.dtsi             | 15 +++++++++++++++
 arch/arm/dts/rcar-x5h-board-u-boot.dts |  3 ++-
 configs/rcar-x5h_board_defconfig       |  1 +
 3 files changed, 18 insertions(+), 1 deletion(-)

diff --git a/arch/arm/dts/r8a78000.dtsi b/arch/arm/dts/r8a78000.dtsi
index 2e6f2f3302..edb697c423 100644
--- a/arch/arm/dts/r8a78000.dtsi
+++ b/arch/arm/dts/r8a78000.dtsi
@@ -88,6 +88,21 @@
 			compatible = "renesas,prr";
 			reg = <0 0xfff00044 0 4>;
 		};
+
+		sysclk: sysclk {
+			compatible = "fixed-clock";
+			#clock-cells = <0x00>;
+			clock-frequency = <0x5f5e100>;
+			clock-output-names = "sysclk";
+		};
+
+		pl011: uart@e5c00000 {
+			compatible = "arm,pl011", "arm,primecell";
+			reg = <0x00 0xe5c00000 0x00 0x1000>;
+			interrupts = <0x00 0x3bf 0x04>;
+			clock-names = "uartclk", "apb_pclk";
+			clocks = <&sysclk>;
+		};
 	};
 
 	timer {
diff --git a/arch/arm/dts/rcar-x5h-board-u-boot.dts b/arch/arm/dts/rcar-x5h-board-u-boot.dts
index 0eeaad15fb..28ca6d336c 100644
--- a/arch/arm/dts/rcar-x5h-board-u-boot.dts
+++ b/arch/arm/dts/rcar-x5h-board-u-boot.dts
@@ -15,11 +15,12 @@
 
 	aliases {
 		serial0 = &hscif0;
+		serial1 = &pl011;
 	};
 
 	chosen {
 		bootargs = "ignore_loglevel rw root=/dev/nfs ip=on";
-		stdout-path = "serial0:921600n8";
+		stdout-path = "serial1:38400n8";
 	};
 
 	memory@48000000 {
diff --git a/configs/rcar-x5h_board_defconfig b/configs/rcar-x5h_board_defconfig
index 8e43bbfc51..4544be6ab1 100644
--- a/configs/rcar-x5h_board_defconfig
+++ b/configs/rcar-x5h_board_defconfig
@@ -22,5 +22,6 @@ CONFIG_REGMAP=y
 CONFIG_SYSCON=y
 CONFIG_CLK=y
 CONFIG_BAUDRATE=921600
+CONFIG_PL01X_SERIAL=y
 CONFIG_SCIF_CONSOLE=y
 
-- 
2.25.1

